editoast/src/views/search.rs

```diff
@@ -692,5 +743,73 @@ pub(super) struct SearchResultItemScenario {
     object(name = "project", config = SearchResultItemProject),
     object(name = "study", config = SearchResultItemStudy),
     object(name = "scenario", config = SearchResultItemScenario),
+    object(name = "trainschedule", config = SearchResultItemTrainSchedule),
 )]
 pub struct SearchConfigFinder;
+
+#[cfg(test)]
+pub mod test {
+
+    use axum::http::StatusCode;
+    use pretty_assertions::assert_eq;
+    use rstest::rstest;
+    use serde_json::json;
+
+    use super::*;
+    use crate::models::fixtures::{create_simple_train_schedule, create_timetable};
+    use crate::models::train_schedule::TrainSchedule;
+    use crate::views::test_app::TestAppBuilder;
+
+    #[rstest]
+    async fn search_trainschedule_post() {
+        let app = TestAppBuilder::default_app();
+        let pool = app.db_pool();
+
+        // Create the timetable in the database
+        let timetable = create_timetable(&mut pool.get_ok()).await;
+        let timetable_id = timetable.id;
+
+        // Load data of the json file
+        #[derive(serde::Deserialize)]
+        struct TrainScheduleInfo {
+            pub train_name: String,
+        }
+        let train_schedule_info: TrainScheduleInfo =
+            serde_json::from_str(include_str!("../tests/train_schedules/simple.json"))
+                .expect("Unable to parse");
+        let train_name = train_schedule_info.train_name;
+
+        // Add a train_schedule in the database
+        let train_schedule: TrainSchedule =
+            create_simple_train_schedule(&mut pool.get_ok(), timetable_id).await;
+        assert_eq!(train_schedule.train_name, train_name);
```

Comment:
```suggestion
        // Add a train_schedule in the database
        let train_schedule: TrainSchedule =
            create_simple_train_schedule(&mut pool.get_ok(), timetable_id).await;
```

I believe `create_simple_train_schedule` already reads this file. I'd say it's not the responsibility of this test to `assert` whether the fixture works as expected or not and that it's fine to assume it does. Wdyt?

Reply:
It thought it was a bad thing to hardcode the expected value as we might want to change the JSON test file one day ? but it surely does look less complicated and in fact this is not the responsability of this script.